Opinfos for avg_pooling

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65941 OpInfo for `*_like` functions
* #64282 OpInfos for some Tensor dtype conversion methods
* **#64214 Opinfos for avg_pooling**

Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

Differential Revision: [D30667797](https://our.internmc.facebook.com/intern/diff/D30667797)

zou3519 committed:
Opinfos for avg_pooling

Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30667797).

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

Differential Revision: [D30667797](https://our.internmc.facebook.com/intern/diff/D30667797)

[ghstack-poisoned]

heitorschueroff commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -7235,6 +7308,40 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
+           dtypes=floating_types(),
+           dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
+           skips=(
+               SkipInfo('TestJit', 'test_variant_consistency_jit'),
```
why is this being skipped?

heitorschueroff reviewed (approved):
LGTM!

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

Differential Revision: [D30667797](https://our.internmc.facebook.com/intern/diff/D30667797)

[ghstack-poisoned]

zou3519 commented:
Re-requesting review because it has been a while since the last review

pmeier commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -7745,6 +7817,40 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
+           dtypes=floating_types(),
+           dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
+           skips=(
+               DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit'),
```
The other `OpInfo`'s always include the dtypes to be skipped here 

```suggestion
               DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit', dtypes=(torch.float32,)),
```

pmeier reviewed (commented):
In general this looks good, but I have a few comments inline to make this more readable.

zou3519 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -3020,6 +3052,46 @@ def generator():
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    # Order: input_shape, kernel_size, stride, padding, ceil_mode, count_include_pad, divisor_override
+    cases = (((1, 3, 9, 5, 9), 3, 1, 1, True, False, 2),
+             ((1, 3, 9, 5, 9), (6, 4, 5), (3, 3, 2), (0, 2, 1), True, True, 2),
+             ((2, 3, 9, 5, 9), (3, 2, 4), (1, 1, 2), 1, False, True, 2),
+             ((1, 3, 9, 5, 9), (6, 3, 4), (3, 3, 3), (0, 1, 0), False, False, 2),
+             ((1, 1, 4, 5, 4), (2, 2, 3), (2, 2, 1), 0, False, True, -2),
+             ((1, 2, 6, 5, 6), (4, 5, 6), (2, 3, 2), 2, True, True, None),
+             ((0, 1, 4, 5, 4), (2, 3, 1), (2, 1, 2), 0, False, True, None))
```
I shrank the inputs

zou3519 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -3020,6 +3052,46 @@ def generator():
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    # Order: input_shape, kernel_size, stride, padding, ceil_mode, count_include_pad
+    cases = (((1, 3, 9), 3, 1, 1, True, False),
+             ((1, 3, 9), (6,), (3,), (2,), True, True),
+             ((2, 3, 9), (3,), (1,), (1,), False, True),
+             ((1, 3, 9), (6,), (3,), (2,), False, False),
+             ((0, 3, 9), (6,), (3,), (2,), False, True))
```
Thanks for the suggestion. This make it a lot easier to read! I've also incorporated the test cases (into the update that is coming soon)

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

Differential Revision: [D30667797](https://our.internmc.facebook.com/intern/diff/D30667797)

[ghstack-poisoned]

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

Differential Revision: [D30667797](https://our.internmc.facebook.com/intern/diff/D30667797)

[ghstack-poisoned]

mruberry reviewed (approved):
Looks good to me but I didn't review the sample inputs thoroughly for coverage (if you'd like to let's take some time briefly together to go over them, @zou3519)

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

Differential Revision: [D30667797](https://our.internmc.facebook.com/intern/diff/D30667797)

[ghstack-poisoned]

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

Differential Revision: [D30667797](https://our.internmc.facebook.com/intern/diff/D30667797)

[ghstack-poisoned]

zou3519 committed:
Update on "Opinfos for avg_pooling"


Added OpInfos for:
- F.adapative_avg_pool{1, 3}d
- F.avg_pool{1, 3}d

The 2d variants already had OpInfos.

Test Plan:
- run tests

Differential Revision: [D30667797](https://our.internmc.facebook.com/intern/diff/D30667797)

[ghstack-poisoned]
